modules/EnsEMBL/Web/Component/Gene/ComparaOrthologs.pm

```diff
@@ -0,0 +1,138 @@
+=head1 LICENSE
+
+Copyright [2009-2024] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package EnsEMBL::Web::Component::Gene::ComparaOrthologs;
+
+use strict;
+use warnings;
+
+sub create_gene_tree_links {
+  my $self = shift;
+  my $params = shift;
+
+  my $hub          = $self->hub;
+  my $species_defs = $hub->species_defs;
+  my $species = $params->{species};
+  my $stable_id = $params->{stable_id};
+  my $orthologue = $params->{orthologue};
+  my $cdb = $params->{cdb};
+
```

Comment:
To get the clusterset_ids by gene, we would need the query gene clusterset_ids as a parameter:
```suggestion
  my $current_gene_clusterset_ids = $params->{current_gene_clusterset_ids};

```

Reply:
> I've made a suggestion in https://github.com/Ensembl/ensembl-webcode/pull/1002 which would make this possible.

I wonder if it's possible to pull this part of your suggestion in the ensembl-webcode PR:

```diff
+  my @current_gene_clusterset_ids;
+  if (exists $species_defs->multi_hash->{'DATABASE_COMPARA'}{'METAZOA_CLUSTERSETS'}) {
+    my @metazoa_clusterset_ids = ('default', 'protostomes', 'insects', 'pangenome_drosophila');  # gene-tree links are displayed in this order
+    @current_gene_clusterset_ids = grep {
+      ($_ eq 'default' && $availability->{'has_gene_tree'})
+      || $availability->{"has_gene_tree_${_}"}
+    } @metazoa_clusterset_ids;
+  }
```
over here, into the metazoa plugin. The reason being that there is a lot of things in that snippet (clusterset ids, `$species_defs->multi_hash->{'DATABASE_COMPARA'}{'METAZOA_CLUSTERSETS'}`, `my @metazoa_clusterset_ids`) that are specific to the metazoa context.

It seems that if we passed `$availability` as a parameter for the `create_gene_tree_links` function, we would be able to pick it up in the metazoa plugin.

Reply:
Fair point @azangru. Yes, AFAICT it would definitely be possible to pass `$availability` as a parameter and use it here to get the `@current_gene_clusterset_ids` instead of having the Metazoa-specific code in the `ensembl-webcode` `ComparaOrthologs` module. 👍